### ISSUE TITLE
feat(transforms): piecewise-linear spline (coupling + marginal)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#70]
## Problem / Request
Add piecewise-linear splines in both coupling (`LinearSplineCoupling`) and marginal (`LinearSplineMarginal`) forms. Each bin is a straight line with learnable heights; derivatives are piecewise-constant. Simpler, cheaper, and more stable than rational-quadratic splines at the cost of lower expressivity.

## User Story
> As a user prototyping a flow on low-dimensional or smooth data, I want a lightweight binned spline that trains faster than the RQ spline and has simpler gradients, so that I can iterate on architectures without paying the RQ-spline cost.

## Motivation
- **Simpler baseline.** Useful for ablations against RQSpline and as a distillation target.
- **Numerical stability.** No rational divisor → no pole-adjacency issues.
- **Low-dim / smooth data.** Enough expressivity when the underlying density is already smooth.
- **Shared infra.** Can share the binned-spline machinery (bin edges, knot heights, softmax-normalised widths) with RQSpline.

## Proposed API
```python
from flowjax.bijections import AbstractBijection

class LinearSplineMarginal(AbstractBijection):
    """Piecewise-linear spline applied elementwise.

    Args:
        n_bins: Number of bins per dim.
        shape: Input shape (d,).
        interval: Bounding interval [-B, B] for the spline support.
                  Outside this interval the map is identity.
    """

class LinearSplineCoupling(AbstractBijection):
    """Coupling layer parameterising per-dim piecewise-linear splines
    from the unchanged half of the input via an MLP conditioner."""
```

## Design Snapshot
```python
# src/gauss_flows/_src/transforms/bijections/elementwise/linear_spline.py
#
# Parameterisation (per dim):
#   widths_i  = softmax(unnormalised_widths) * 2B   ∈ R_+^K, sum to 2B
#   heights_i = softmax(unnormalised_heights) * 2B  ∈ R_+^K, sum to 2B
#   knots_x = cumsum(widths),   knots_y = cumsum(heights)  (both offset by -B)
#
# Forward: find bin k containing x, compute
#   y = knots_y[k-1] + (x - knots_x[k-1]) * (heights_k / widths_k)
#   log|dy/dx| = log(heights_k / widths_k)
#
# Outside [-B, B]: identity (y = x, log_det = 0).
# Share bin-search / transform-indexing utilities with the existing RQSpline
# implementation — consider refactoring a shared `_binned.py` helper module.
```

## Mathematical Notes
```text
For x in bin k = [knots_x[k-1], knots_x[k]]:
    slope_k = heights_k / widths_k
    y = knots_y[k-1] + (x - knots_x[k-1]) * slope_k
    log|dy/dx| = log(slope_k)   (piecewise constant in x, exact)

Inverse: identical structure with roles of x and y swapped (linear in each
bin, closed form — no Newton / bisection).
```

## References & Existing Code
- Paper: Müller et al. 2019. *Neural Importance Sampling*. ACM TOG (introduces piecewise-linear and piecewise-quadratic coupling).
- Reference impl: `vislearn/FrEIA/FrEIA/modules/splines/linear.py::LinearSpline`
- gauss_flows analogue: `src/gauss_flows/_src/transforms/bijections/coupling/spline.py::RQSplineCoupling` and `bijections/elementwise/spline.py::RQSplineMarginal`

## Implementation Steps
- [ ] Add `LinearSplineMarginal` in `src/gauss_flows/_src/transforms/bijections/elementwise/linear_spline.py`
- [ ] Add `LinearSplineCoupling` in `src/gauss_flows/_src/transforms/bijections/coupling/linear_spline.py`
- [ ] Optional: extract shared binned-spline utilities into `_src/transforms/bijections/_binned.py` and refactor RQSpline to use them
- [ ] Re-export from each `__init__.py` up to top-level
- [ ] Add `__all__` entries

## Definition of Done
- [ ] Code at intended paths
- [ ] Public API exported
- [ ] Tests pass
- [ ] Lint + typecheck pass

## Testing
- [ ] Round-trip: `inverse(transform(x)) == x` inside and outside `[-B, B]`
- [ ] Identity outside `[-B, B]`: `transform(x) == x`, `log_det == 0` for `|x| > B`
- [ ] At default init (equal widths = equal heights): map is identity on `[-B, B]`
- [ ] Log-det = log(slope_k) matches finite-difference Jacobian
- [ ] Coupling variant: conditioner gradient flows only from unchanged → transformed half

## Documentation
- [ ] API reference entries for both classes
- [ ] Note in docstring comparing vs RQSpline (cheaper, less expressive)

## Relationships
- Related: FrEIA port series
